Client/Helpers/FxALoginHelper.swift

```diff
@@ -136,10 +136,10 @@ class FxALoginHelper {
     // It manages the asking for user permission for notification and registration 
     // for APNS and WebPush notifications.
     func application(_ application: UIApplication, didReceiveAccountJSON data: JSON) {
-        if data["keyFetchToken"].stringValue() == nil || data["unwrapBKey"].stringValue() == nil {
+        if data["keyFetchToken"] == JSON.null || data["unwrapBKey"] == JSON.null {
             // The /settings endpoint sends a partial "login"; ignore it entirely.
             log.error("Ignoring didSignIn with keyFetchToken or unwrapBKey missing.")
-            return self.loginDidFail()
+            return
         }
```

Comment:
I'm not sure I understand this PR.

This code runs just as the user successfully logs in to FxA. The changes appear to leave the `FxALoginHelper` in an inconsistent state?

Reply:
Sorry about that, updated comment. Using `self.loginDidFail()` was causing the content view controller to be automatically [dismissed](https://github.com/mozilla-mobile/firefox-ios/blob/0120d8a69466b5c097a68cef54f092b25f8bbee9/Client/Frontend/Settings/FxAContentViewController.swift#L250). Not having it allows the view to appear and waits for the account to be verified.